StRoot/StEvent/StFcsHit.cxx

```diff
@@ -139,3 +141,20 @@ void StFcsHit::print(Option_t *option) const {
     }
     cout << endl;
 }
+
+void StFcsHit::addGeantTrack(unsigned int id, float e){    
+    unsigned int n=mGeantTracks.size();
+    for(unsigned int i=0; i<n; i++){
+	if(mGeantTracks[i].first == id) {mGeantTracks[i].second += e; return;}
+    }
```

Comment:
auto trk = std::find_if(mGeantTracks.begin(), mGeantTracks.end(),
                            [&id](const std::pair<unsigned int, float>& t) {
                                return t.first == id;});
    if(trk!=mGeantTracks.end()) {trk->second += e; return;}
This seems to work...and while I do use lambda expressions elsewhere... I try to avoid it when possible. Maybe my inexperience, but it makes much harder to understand the code. In this case I don't think it made code any more better nor generic nor easier to maintain. If you insist, I can switch...

Reply:
I mentioned one of the reasons above. 
It might be easier to read if you assign the comparison function to a variable. C++20 provides an even terser syntax

Reply:
Are we using c++11? 17? 20?